docs/source/api/commands.rst

```diff
@@ -1,47 +1,626 @@
+
+    "Account ID", "ID of account where role will be deleted", "already existent", "makoto@soramitsu"
+    "Role name", "a detached role name", "existing role", "User"
+
+Validation
+^^^^^^^^^^
+
+1. The role exists in the system
+2. The account has such role
+
 Grant permission
 ----------------
 
+Purpose
+^^^^^^^
+
+The purpose of grant permission command is to give another account rights to perform actions on the account of transaction sender (give someone right to do something with my account).
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message GrantPermission {
+       string account_id = 1;
+       string permission_name = 2;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+    "Account ID", "id of account whom rights are granted", "already existent", "makoto@soramitsu"
+    "Permission name", "name of granted permission", "permission is defined", "CanTransferAssets"
+
+
+Validation
+^^^^^^^^^^
+
+1. Account exists
+2. Transaction creator is allowed to grant this permission
+
+Remove signatory
+----------------
+
+Purpose
+^^^^^^^
+
+Purpose of remove signatory command is to remove a public key, associated with an identity, from an account
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message RemoveSignatory {
+        string account_id = 1;
+        bytes public_key = 2;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+    "Account ID", "id of account whom rights are granted", "already existent", "makoto@soramitsu"
+    "Public key", "Signatory to delete", "ed25519 public key", "407e57f50ca48969b08ba948171bb2435e035d82cec417e18e4a38f5fb113f83"
+
+Validation
+^^^^^^^^^^
+
+1. When signatory is deleted, we should check if invariant of **size(signatories) >= quorum** holds
+2. Signatory should have been previously added to the account
+
+Two cases:
+
+    Case 1. When transaction creator wants to remove signatory from their account and he or she has permission CanRemoveSignatory
+
+    Case 2. CanRemoveSignatory was granted to transaction creator
+
 Revoke permission
 -----------------
 
+Purpose
+^^^^^^^
+
+The purpose of revoke permission command is to revoke or dismiss given granted permission from another account in the network.
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message RevokePermission {
+       string account_id = 1;
+       string permission_name = 2;
+    }
+
+Structure
+^^^^^^^^^
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+        "Account ID", "id of account whom rights are granted", "already existent", "makoto@soramitsu"
+        "Permission name", "name of granted permission", "permission was granted", "CanTransferAssets"
+
+Validation
+^^^^^^^^^^
+
+1. Transaction creator should have previously granted this permission to a target account
```

Comment:
I suggest removing the number. It makes me think that document creator just forgot to list the rest of elements.